Assets/Scripts/NPCs/BasicNurse.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using NPCs.Components;
+using UnityEngine;
+
+namespace NPCs
+{
+    [RequireComponent(typeof(Chaser))]
+    [RequireComponent(typeof(Responder))]
+    [RequireComponent(typeof(Searcher))]
+    [RequireComponent(typeof(Patroller))]
+    [RequireComponent(typeof(Waiter))]
+    [RequireComponent(typeof(Animator))]
+    public class BasicNurse : BaseAi
+    {
+        private static readonly HashSet<string> States = new HashSet<string>()
```

Comment:
Enums don't work with inheritance; the alternative is to declare one master enum of possible states in BaseAi and keep adding to it whenever we make new NPC types, but I thought that defeated the purpose of having inheritance in the first place.

Reply:
One thing maybe you can check out is the way I did the UI system with inheritance - basically where we can define a unique enum outside this class, and then make the BaseAI class templated on this enum. Check out `PauseMenuManager` for example! I think that may be cleaner personally.

Reply:
Working on this now - I'm changing up all the NPC components so that instead of directly modifying the BasicNurse's behaviors (e..g changing state or destination), they just trigger events to the BasicNurse when important things happen

Reply:
That's awesome, I really like that. Looking forward to see that design get built up!!
Thanks for taking the time to refactor this all man